Add generic support for WM8804 Digi boards/HATs using hifiberry_digi/wm8804 driver combination

clivem opened the pull request:
Modify hifiberry_digi driver and add dedicated dt overlays for Geekroo and IQAudIO Digi products

pelwell commented on sound/soc/bcm/hifiberry_digi.c:
```diff
@@ -148,8 +148,6 @@ static struct snd_soc_ops snd_rpi_hifiberry_digi_ops = {
 static struct snd_soc_dai_link snd_rpi_hifiberry_digi_dai[] = {
 {
-	.name		= "HifiBerry Digi",
-	.stream_name	= "HifiBerry Digi HiFi",
```
I see no harm in leaving these here as defaults.

pelwell commented on sound/soc/bcm/hifiberry_digi.c:
```diff
@@ -163,33 +161,47 @@ static struct snd_soc_dai_link snd_rpi_hifiberry_digi_dai[] = {
 
 /* audio machine driver */
 static struct snd_soc_card snd_rpi_hifiberry_digi = {
-	.name         = "snd_rpi_hifiberry_digi",
```
Ditto.

pelwell commented on sound/soc/bcm/hifiberry_digi.c:
```diff
@@ -163,33 +161,47 @@ static struct snd_soc_dai_link snd_rpi_hifiberry_digi_dai[] = {
+
+	    snprintf(prop, sizeof(prop), "%scard-name", prefix);
+	    if (of_property_read_string(np, prop, &card->name))
+		card->name = "snd_rpi_hifiberry_digi";
```
The comments for of_property_read_string say:

```
 * @out_string: pointer to null terminated return string, modified only if
 *              return value is 0.
```

If you leave the default strings in snd_rpi_hifiberry_digi & _dai then you can replace these conditionals with simple calls to of_property_read_string and get the same result.

pelwell commented on sound/soc/bcm/hifiberry_digi.c:
```diff
@@ -202,7 +214,9 @@ static int snd_rpi_hifiberry_digi_remove(struct platform_device *pdev)
 static const struct of_device_id snd_rpi_hifiberry_digi_of_match[] = {
+	{ .compatible = "geekaroo,wm8804-digi", },
 	{ .compatible = "hifiberry,hifiberry-digi", },
+	{ .compatible = "iqaudio,wm8804-digi", },
```
Adding compatible strings for known users seems wrong to me - this is now meant to be generic driver. Either add a second vendor-neutral compatible string or make all overlays use "hifiberry,hifiberry-digi", which seems like a reasonable tariff for using the Hifiberry driver.

pelwell commented on arch/arm/boot/dts/overlays/README:
```diff
@@ -308,6 +308,12 @@ Params: int_pin                 GPIO used for INT (default 25)
 
+Name:   geekroo-digi-wm8804-audio
+Info:   Configures the Geekroo Digi WM8804 audio card
+Load:   dtoverlay=geekroo-digi
```
This should say `dtoverlay=geekroo-digi-wm8804-audio`.

pelwell commented on arch/arm/boot/dts/overlays/README:
```diff
@@ -546,6 +552,12 @@ Params: 24db_digital_gain       Allow gain to be applied via the PCM512x codec
 
+Name:   iqaudio-digi-wm8804-audio
+Info:   Configures the IQAudIO Digi WM8804 audio card
+Load:   dtoverlay=iqaudio-digi
```
This should say `dtoverlay=iqaudio-digi-wm8804-audio`.

clivem commented:
@pelwell Don't know if you got a ping yesterday when I pushed those updates, so just letting you know that I did. ;)

pelwell commented:
No I didn't - I'll take a look.

pelwell commented:
That looks much neater. I would argue that only the hifiberry overlay needs parameters to set the device names (or even a generic version of the overlay), and that they should get a brief mention in the README.

clivem commented:
OK, Phil, all I'm really bothered about here at the moment is adding support for Gordons digital output board. I think what I'm going to do, is cut and paste hifiberry_digi.c into iqaudio_digi.c. (It makes more sense to do that now without the unique compatible string, anyway.)

pelwell commented:
Whoah there - if you have a good reason for wanting a unique compatible string, make one. Otherwise I'll just remove the parameter declarations from the other overlays and push the commits.

DigitalDreamtime committed:
Add IQAudIO Digi WM8804 board support

Support IQAudIO Digi board with iqaudio_digi machine driver and
 iqaudio-digi-wm8804-audio overlay.

NB. Machine driver is a cut and paste of hifiberry_digi code, with format
    and general cleanup to comply with kernel coding standards.

Signed-off-by: DigitalDreamtime <clive.messer@digitaldreamtime.co.uk>

pelwell commented:
checkpatch reports two errors and a bunch of warnings. I'm not bothered about the lack of documentation, but the other style issues should be fixed.

clivem commented:
Grrrr.....

Would you like me to submit a patch that sorts out other peoples failure to comply with the output of checkpatch, or does the checkpatch diligence requirement only apply to me? ;)

```
ERROR: open brace '{' following function declarations go on the next line
#43: FILE: justboom-digi.c:40:
+static int snd_rpi_justboom_digi_startup(struct snd_pcm_substream *substream) {

WARNING: Missing a blank line after declarations
#47: FILE: justboom-digi.c:44:
+   struct snd_soc_codec *codec = rtd->codec;
+   snd_soc_update_bits(codec, WM8804_PWRDN, 0x3c, 0x00);

WARNING: line over 80 characters
#51: FILE: justboom-digi.c:48:
+static void snd_rpi_justboom_digi_shutdown(struct snd_pcm_substream *substream) {

ERROR: open brace '{' following function declarations go on the next line
#51: FILE: justboom-digi.c:48:
+static void snd_rpi_justboom_digi_shutdown(struct snd_pcm_substream *substream) {

WARNING: Missing a blank line after declarations
#55: FILE: justboom-digi.c:52:
+   struct snd_soc_codec *codec = rtd->codec;
+   snd_soc_update_bits(codec, WM8804_PWRDN, 0x3c, 0x3c);

ERROR: spaces required around that '=' (ctx:VxV)
#68: FILE: justboom-digi.c:65:
+   long mclk_freq=0;
                  ^

ERROR: spaces required around that '=' (ctx:VxV)
#69: FILE: justboom-digi.c:66:
+   int mclk_div=1;
                ^

ERROR: spaces required around that '=' (ctx:VxV)
#70: FILE: justboom-digi.c:67:
+   int sampling_freq=1;
                     ^

ERROR: spaces required around that '<=' (ctx:VxV)
#76: FILE: justboom-digi.c:73:
+   if (samplerate<=96000) {
                  ^

ERROR: spaces required around that '=' (ctx:VxV)
#77: FILE: justboom-digi.c:74:
+       mclk_freq=samplerate*256;
                 ^

ERROR: spaces required around that '=' (ctx:VxV)
#78: FILE: justboom-digi.c:75:
+       mclk_div=WM8804_MCLKDIV_256FS;
                ^

ERROR: spaces required around that '=' (ctx:VxV)
#80: FILE: justboom-digi.c:77:
+       mclk_freq=samplerate*128;
                 ^

ERROR: spaces required around that '=' (ctx:VxV)
#81: FILE: justboom-digi.c:78:
+       mclk_div=WM8804_MCLKDIV_128FS;
                ^

ERROR: switch and case should be at the same indent
#84: FILE: justboom-digi.c:81:
+   switch (samplerate) {
+       case 32000:
[...]
+       case 44100:
[...]
+       case 48000:
[...]
+       case 88200:
[...]
+       case 96000:
[...]
+       case 176400:
[...]
+       case 192000:
[...]
+       default:

ERROR: spaces required around that '=' (ctx:VxV)
#86: FILE: justboom-digi.c:83:
+           sampling_freq=0x03;
                         ^

ERROR: spaces required around that '=' (ctx:VxV)
#89: FILE: justboom-digi.c:86:
+           sampling_freq=0x00;
                         ^

ERROR: spaces required around that '=' (ctx:VxV)
#92: FILE: justboom-digi.c:89:
+           sampling_freq=0x02;
                         ^

ERROR: spaces required around that '=' (ctx:VxV)
#95: FILE: justboom-digi.c:92:
+           sampling_freq=0x08;
                         ^

ERROR: spaces required around that '=' (ctx:VxV)
#98: FILE: justboom-digi.c:95:
+           sampling_freq=0x0a;
                         ^

ERROR: spaces required around that '=' (ctx:VxV)
#101: FILE: justboom-digi.c:98:
+           sampling_freq=0x0c;
                         ^

ERROR: spaces required around that '=' (ctx:VxV)
#104: FILE: justboom-digi.c:101:
+           sampling_freq=0x0e;
                         ^

ERROR: space required after that ',' (ctx:VxV)
#132: FILE: justboom-digi.c:129:
+   return snd_soc_dai_set_bclk_ratio(cpu_dai,64);
                                             ^

ERROR: code indent should use tabs where possible
#138: FILE: justboom-digi.c:135:
+        .startup = snd_rpi_justboom_digi_startup,$

WARNING: please, no spaces at the start of a line
#138: FILE: justboom-digi.c:135:
+        .startup = snd_rpi_justboom_digi_startup,$

ERROR: code indent should use tabs where possible
#139: FILE: justboom-digi.c:136:
+        .shutdown = snd_rpi_justboom_digi_shutdown,$

WARNING: please, no spaces at the start of a line
#139: FILE: justboom-digi.c:136:
+        .shutdown = snd_rpi_justboom_digi_shutdown,$

WARNING: suspect code indent for conditional statements (8, 12)
#171: FILE: justboom-digi.c:168:
+   if (pdev->dev.of_node) {
+       struct device_node *i2s_node;

WARNING: Missing a blank line after declarations
#174: FILE: justboom-digi.c:171:
+       struct snd_soc_dai_link *dai = &snd_rpi_justboom_digi_dai[0];
+       i2s_node = of_parse_phandle(pdev->dev.of_node,

WARNING: suspect code indent for conditional statements (12, 24)
#177: FILE: justboom-digi.c:174:
+       if (i2s_node) {
+           dai->cpu_dai_name = NULL;

```

pelwell commented:
Sorry - it was your commit message that prompted the check:

```
NB. Machine driver is a cut and paste of hifiberry_digi code, with format
    and general cleanup to comply with kernel coding standards.
```

clivem commented:
> Sorry - it was your commit message that prompted the check:

Well, I had already fixed-up most of what checkpatch flagged. I missed a couple.......

pelwell commented:
Will you be fixing them, or are you happy for it to go in as it is?

clivem commented:
> Will you be fixing them, or are you happy for it to go in as it is?

Eh, I already had updated the patch to take care of the two items still being flagged as formatting "ERRORS", before I posted the "Grrr...." message. The only things left, were the warnings over documentation and undocumented dt strings, which I thought you said you were not bothered about.

shawaj commented:
Only just saw this. @clivem @pelwell do you want us to submit a fix for any errors in ours?

Sorry if it's not up to scratch. Let me know what you want doing and will sort
